test_runner/src/test/kotlin/ftl/run/platform/android/CreateAndroidTestContextKtTest.kt

```diff
@@ -292,4 +292,25 @@ class CreateAndroidTestContextKtTest {
             assertTrue(customShards.shards.values.containsAll(context.shards.map { it.testMethodNames }))
         }
     }
+
+    @Test
+    fun `Annotated parameterized tests cases should run when specified in test-targets`() {
+        val testInstrumentationContext = InstrumentationTestContext(
+            FileReference("", ""),
+            FileReference("../test_projects/android/apks/parametrized-app-debug-androidTest.apk", ""),
+            args = args
+        )
+
+        val parsedTests = testInstrumentationContext
+            .getFlankTestMethods(
+                TestFilter(
+                    "allOf [withAnnotation (flank.shard.annotationbug.Annotation), anyOf []]",
+                    shouldRun = { true })
+            )
+            .map { it.testName }
+            .toSet()
+
+        assertTrue(parsedTests.isNotEmpty())
+        assertTrue(parsedTests.contains("class flank.shard.annotationbug.ParameterizedAnnotatedTest#useAppContext"))
+    }
```

Comment:
Looks great, can we have another Test with an annotated test with a 'wrong' annotation so we can check it doesn't get included?